include/openmc/mgxs_interface.h

```diff
@@ -12,36 +12,71 @@
 namespace openmc {
 
 //==============================================================================
-// Global variables
+// Global MGXS data container structure
 //==============================================================================
 
-namespace data {
+struct MgxsInterface
+{
+  int num_energy_groups;
+  int num_delayed_groups;
 
-extern std::vector<Mgxs> nuclides_MG;
-extern std::vector<Mgxs> macro_xs;
-extern int num_energy_groups;
-extern int num_delayed_groups;
-extern std::vector<double> energy_bins;
-extern std::vector<double> energy_bin_avg;
-extern std::vector<double> rev_energy_bins;
+  // List of available names in the HDF5 file
+  std::vector<std::string> xs_names;
+  std::vector<std::string> xs_to_read;
+  std::vector<std::vector<double>> xs_temps_to_read;
 
-} // namespace data
+  // Name of the HDF5 file which contains mgxs
+  std::string cross_sections_path;
 
-//==============================================================================
-// Mgxs data loading interface methods
-//==============================================================================
+  std::vector<Mgxs> nuclides_MG;
+  std::vector<Mgxs> macro_xs;
+
+  std::vector<double> energy_bins;
+  std::vector<double> energy_bin_avg;
+  std::vector<double> rev_energy_bins;
```

Comment:
Data members in a class should have a [trailing underscore](https://docs.openmc.org/en/latest/devguide/styleguide.html#naming).

Reply:
Hm, OK, changed